src/components/RefinementList.vue

```diff
@@ -0,0 +1,150 @@
+        <ais-search-input v-model="searchForFacetValues"></ais-search-input>
+      </div>
+      <ul :class="suit('list')">
+        <li
+          :class="[suit('item'), {[suit('item', 'selected')]: item.isRefined}]"
+          v-for="item in state.items"
+          :key="item.value"
+        >
+          <slot name="item" :item="item" :refine="state.refine">
+            <label :class="suit('label')">
+              <input
+                :class="suit('checkbox')"
+                type="checkbox"
+                :value="item.value"
+                :checked="item.isRefined"
+                @change="state.refine(item.value)"
+              />
+              <span v-if="searchable" :class="suit('labelText')">
+                <ais-highlight attribute="item" :hit="transformItemToHit(item)"/>
+              </span>
+              <span v-else :class="suit('labelText')">{{item.label}}</span>
+              <span :class="suit('count')">{{item.count}}</span>
+            </label>
+          </slot>
+        </li>
+      </ul>
+      <button
+        :class="[suit('showMore'), { [suit('showMore', 'disabled')]: !state.canToggleShowMore }]"
+        @click="toggleShowMore"
+        v-if="showMore"
+        :disabled="!state.canToggleShowMore"
+      >
+        <slot name="showMoreTitle" :is-showing-more="state.isShowingMore">
+          Show {{state.isShowingMore ? 'less' : 'more'}}
+        </slot>
+      </button>
+    </slot>
+  </div>
+</template>
+
+<script>
+import algoliaComponent from '../component';
+import { connectRefinementList } from 'instantsearch.js/es/connectors';
+import AisSearchInput from './SearchInput';
+import AisHighlight from './Highlight';
+
+const noop = () => {};
+
+export default {
+  components: { AisSearchInput, AisHighlight },
+  mixins: [algoliaComponent],
+  props: {
+    attribute: {
+      type: String,
+      required: true,
+    },
+    searchable: {
+      type: Boolean,
+      default: false,
+    },
+    operator: {
+      type: ['and', 'or'],
+      required: false,
+    },
+    limit: {
+      type: Number,
+      required: false,
+    },
+    showMoreLimit: {
+      type: Number,
+      default: 20,
+      required: false,
+    },
+    showMore: {
+      type: Boolean,
+      default: false,
+      required: false,
+    },
+    sortBy: {
+      type: [Array | Function],
+      required: false,
+    },
+    transformItems: {
+      type: Function,
+      required: false,
+    },
+  },
+  data() {
+    return {
+      widgetName: 'RefinementList',
+      searchForFacetValuesQuery: '',
+    };
+  },
+  beforeCreate() {
+    this.connector = connectRefinementList;
+  },
+  computed: {
+    searchForFacetValues: {
+      get() {
+        return this.searchForFacetValuesQuery;
+      },
+      set(value) {
+        this.state.searchForItems(value);
+        this.searchForFacetValuesQuery = value;
+      },
+    },
+    toggleShowMore() {
+      return this.state.toggleShowMore || noop;
```

Comment:
We need this because of the IS issue? To avoid the value to be undefined?

Reply:
yes, if the value is undefined (even when v-if is false) it gives a warning to the console, so I need something which is always callable